lib/route/qdisc.c

```diff
@@ -403,6 +403,38 @@ struct rtnl_qdisc *rtnl_qdisc_get_by_parent(struct nl_cache *cache,
 	return NULL;
 }
 
+/**
+ * Search qdisc by kind
+ * @arg cache		Qdisc cache
+ * @arg ifindex		Interface index
+ * @arg kind		Qdisc kind (tbf, htb, cbq, etc)
+ *
+ * Searches a qdisc cache previously allocated with rtnl_qdisc_alloc_cache()
+ * and searches for a qdisc matching the interface index and kind.
+ *
+ * The reference counter is incremented before returning the qdisc, therefore
+ * the reference must be given back with rtnl_qdisc_put() after usage.
+ *
+ * @return pointer to qdisc inside the cache or NULL if no match was found.
```

Comment:
I am not very happy about simple lookup functions increasing the reference count. On the other hand, that is consistent with `rtnl_qdisc_get_by_parent()`, so I guess it makes sense.

Reply:
I understand your reasoning, and leaving this line as it is.